include/circt/Dialect/Arc/ArcOps.td

```diff
@@ -452,6 +460,20 @@ def PassThroughOp : ArcOp<"passthrough", [NoTerminator, NoRegionArguments]> {
   }];
 }
 
+def InitialOp : ArcOp<"initial", [RecursiveMemoryEffects, NoTerminator, NoRegionArguments]> {
```

Comment:
As mentioned above, I shortly had it as isolated from above. But for sake of simplicity and to keep it in line with the PassthroughOp I decided not to keep that. What would be the benefit of making it isolated form above?

Reply:
Since you just compute the initial values in this region it doesn't make sense to use SSA values from arbitrary places in the design. Making it isolated provides additional compile-time guarantees. I'd just leave it as is for now. 

We are still thinking about how we can better represent the storage rather than hardcoding integer offsets. Maybe we'll unify the alloc ops and storage argument at some point (I have an experimental branch doing that, but it makes use of symbols which is a bit of an issue in verifier time). If something like that makes it in, one block argument would be enough and we could make it isolated from above, but at that point it will be just a small change.